numba/tests/test_typeof.py

```diff
@@ -301,6 +308,77 @@ def test_omitted_args(self):
         self.assertEqual(len({ty0, ty1, ty2}), 3)
         self.assertEqual(ty3, ty2)
 
+    def test_builtin_types(self):
+        self.assertEqual(typeof(int), typeof(0))
+        self.assertEqual(typeof(float), typeof(0.0))
+        self.assertEqual(typeof(complex), typeof(complex(0)))
+        self.assertEqual(typeof(str), typeof("numba"))
```

Comment:
It's not clear to me that `typeof(x) == typeof(type(x))` makes any sense.
I'd expect these to return `numba.types.TypeRef`, if not the `ValueError` they give today.

Would it make sense to introduce a new `as_numba_type` function?

Reply:
In fact, this change is incompatible with the behavior today:
```
In [22]: numba.typeof(np.uint8(1))
Out[22]: uint8

In [23]: numba.typeof(type(np.uint8(1)))
Out[23]: class(uint8)
```

Reply:
I had a reason for making it part of `typeof` originally, but can't remember it.  Making a new `as_numba_type` gives a cleaner API and offers some cool opportunities for user extensions.